viewer/packages/tools/src/Timeline/TimelineTool.ts

```diff
@@ -0,0 +1,174 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import { Cognite3DModel } from '@reveal/core';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+import { TimelineKeyframe } from './TimelineKeyframe';
+import { NodeCollectionBase, NodeAppearance } from '@reveal/core/src';
+
+/**
+ * Tool to applying styles to nodes based on date to play them over in Timeline
+ */
+export class TimelineTool extends Cognite3DViewerToolBase {
+  private readonly _model: Cognite3DModel;
+  private _timelineKeyframes: TimelineKeyframe[];
+  private _intervalId: any = 0;
+  private _currentDate: Date;
+  private _allDates: Date[];
+
+  constructor(cadModel: Cognite3DModel) {
+    super();
+
+    this._model = cadModel;
+    this._timelineKeyframes = new Array<TimelineKeyframe>();
+    this._allDates = new Array<Date>();
+    this._currentDate = new Date();
+  }
+
+  /**
+   * Create Key frame for the Timeline
+   * @param date - date value by Date.now() since January 1, 1970
+   * @param nodeCollection - Set of nodes to be rendered for the TimelineKeyframe
+   * @param nodeAppearance - Styling of the nodes
+   */
+  public createKeyFrame(date: Date, nodeCollection: NodeCollectionBase, nodeAppearance: NodeAppearance) {
```

Comment:
Reopening - the current API is different from the API I proposed in the last comment.

As mentioned, I don't think it's very relevant to have the ability to create keyframes with a nodeCollection and nodeAppearance directly. It will be a two step process:
1. Create keyframe
2. Assign styles to the created keyframe